src/topology/metric_space/pi_nat.lean

```diff
@@ -193,6 +193,64 @@ lemma update_mem_cylinder (x : Π n, E n) (n : ℕ) (y : E n) :
   update x n y ∈ cylinder x n :=
 mem_cylinder_iff.2 (λ i hi, by simp [hi.ne])
 
+section res
+
+variable {α : Type*}
+open list
+
+/-- In the case where `E` has constant value `α`,
+the cylinder `cylinder x n` can be identified with the element of `list α`
+consisting of the first `n` entries of `x`. See `cylinder_eq_res`.
+We call this list `res x n`, the restriction of `x` to `n`.-/
+def res (x : ℕ → α) : ℕ → list α
+  | 0            := nil
+  | (nat.succ n) := x n :: res n
+
+@[simp] lemma res_zero (x : ℕ → α) : res x 0 = @nil α := rfl
+@[simp] lemma res_succ (x : ℕ → α) (n : ℕ) : res x n.succ = x n :: res x n := rfl
+
+@[simp] lemma res_length (x : ℕ → α) (n : ℕ) : (res x n).length = n :=
+begin
+  induction n with n ih,
+  { refl },
+  simp[ih],
+end
+
+/-- The restrictions of `x` and `y` to `n` are equal if and only if `x m = y m` for all `m < n`.-/
+lemma res_eq_res (x y : ℕ → α) (n : ℕ) : res x n = res y n ↔ ∀ m < n, x m = y m :=
```

Comment:
Iffs get less explicit arguments so that they are easier to use one way. Also, semi-implicit arguments are useful when you have a forall appearing in a lemma statement and would make it an argument implicit were the forall a lemma on its own.
```suggestion
lemma res_eq_res {x y : ℕ → α} {n : ℕ} : res x n = res y n ↔ ∀ ⦃m⦄, m < n → x m = y m :=
```
Here's what I mean, with the forall on its own (don't add it! just an example)
```
lemma res_eq_res {x y : ℕ → α} {n : ℕ} (h : res x n = res y n) {m} : m < n → x m = y m :=
```